crates/bitwarden-sm/src/generators/secret_generator.rs

```diff
@@ -0,0 +1,83 @@
+use bitwarden_generators::{password, PasswordError, PasswordGeneratorRequest};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum GenerateSecretError {
+    #[error("No character set enabled")]
+    NoCharacterSetEnabled,
+    #[error("Invalid secret length")]
+    InvalidLength,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct GenerateSecretRequest {
+    /// Include lowercase characters (a-z).
+    pub include_lowercase: bool,
+    /// Include uppercase characters (A-Z).
+    pub include_uppercase: bool,
+    /// Include numbers (0-9).
+    pub include_numbers: bool,
+    /// Include special characters: ! @ # $ % ^ & *
+    pub include_special: bool,
+
+    /// The length of the generated secret.
+    /// Note that the secret length must be greater than the sum of all the minimums.
+    pub length: u8,
+
+    /// When set to true, the generated secret will not contain ambiguous characters.
+    /// The ambiguous characters are: I, O, l, 0, 1
+    pub avoid_ambiguous: bool, // TODO: Should we rename this to include_all_characters?
```

Comment:
```suggestion
    pub avoid_ambiguous: bool,
```
💭 
If I'm understanding this comment correctly, `include_all_characters` would be the inverse of `avoid_ambigous`?

If `include_all_characters` is set to `true` I would expect `I, O, l, 0, 1` to be used.
If `avoid_ambigous` is set to `true` I would expect `I, O, l, 0, 1` **NOT** to be used.

My thought is to leave it the same as the `PasswordGeneratorRequest`.

Regardless of what we go with, the `TODO` should be removed to avoid future confusion.